src/Formatters/FormatterTrait.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Drush\Formatters;
+
+use Consolidation\Filter\FilterOutputData;
+use Consolidation\Filter\LogicalOpFactory;
+use Drush\Attributes\FilterDefaultField;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+trait FormatterTrait
+{
+    /**
+     * Adds options to the command definition based on data type. The --format description is dynamic.
+     *
+     * @param string $dataType
+     *   Usually the same as the return type of a doExecute() method.
+     */
+    public function addFormatterOptions(string $dataType): void
+    {
+        $inputOptions = $this->formatterManager->automaticOptions($this->getFormatterOptions(), $dataType);
```

Comment:
I think this can be refactored to avoid the intermediate instances and private field access. This can wait until moving this code into output-formatter.